tests/unit/test_calibration.py

```diff
@@ -123,27 +123,80 @@ def test_closest_point_fit() -> None:
     ), f"Offset is {linear_transform.offset} expected {expected_offset})"
 
 
-def test_calibrate_calibration_error() -> None:
-    """Test error when calibration was not possible."""
+def test_closest_point_fit_constrainedmin() -> None:
+    """Test closest point fit."""
+    scale = np.array([1.1, 0.9, 0.6])
+    offset = np.array([0.1, 0.2, 0.1])
+    expected_scale = 1 / scale
+    expected_offset = -offset / scale
+    data = np.random.randn(1000, 3)
+    norms = np.linalg.norm(data, axis=1, keepdims=True)
+    unit_sphere = data / norms
+    calibrator = calibration.ConstraintedMinimizationCalibration()
+
+    linear_transform = calibrator._closest_point_fit((unit_sphere * scale) + offset)
+
+    assert np.allclose(
+        linear_transform.scale, expected_scale, atol=1e-3
+    ), f"Scale is {linear_transform.scale} expected {expected_scale}"
+    assert np.allclose(
+        linear_transform.offset, expected_offset, atol=1e-3
+    ), f"Offset is {linear_transform.offset} expected {expected_offset})"
+
+
+def test_closest_point_fit_constrainmin_Calibration_Error() -> None:
+    """Test closest point fit raises Calibration Error for constrained min."""
+    data = np.random.randn(1000, 3)
+    calibrator = calibration.ConstraintedMinimizationCalibration(max_iterations=0)
+
+    with pytest.raises(calibration.CalibrationError):
+        calibrator._closest_point_fit(data)
+
+
+def test_constrainmin_Calibration_Error() -> None:
+    """Test ConstrainedMnimization Calibration raises Calibration Error."""
     dummy_measure = create_dummy_measurement(
         sampling_rate=1,
         duration_hours=1,
     )
-    calibrator = calibration.Calibration(
-        min_calibration_error=0.0001, max_iterations=5, min_acceleration=0
+    calibrator = calibration.ConstraintedMinimizationCalibration(
+        min_calibration_error=0, max_iterations=1
+    )
+
+    with pytest.raises(calibration.CalibrationError):
+        calibrator.run_calibration(dummy_measure)
+
+
+def test_calibrate_calibration_error() -> None:
+    """Test error when calibration was not possible."""
```

Comment:
Does this need the full set-up when you're only testing for a CalibrationError?

Reply:
I don't think I touched this test, but I based the tests for the new class as clones of the existing GGIR ones. If you mean the function above, I was trying to get the second CalibrationError that we try to Raise explicitly in `run_calibration` (this is actually the line that doesn't seem to be covered) but it might also change based on your comments about organizing our `run_calibration`

Reply:
it shows as modified in the PR :s, I was referring to test_calibrate_calibration_error

Reply:
The only thing I changed in that one is the naming for the new `GGIRCalibration` class. I believe the `CalibrationError` only gets raised in `calibrator._calibrate` so it needs the full `dummy_measure`